### PR TITLE
Expand parsing tests

### PR DESCRIPTION
**Pull request by schlosna**
Add a few specific edge case tests in addition to property based tests for `Deadlines.tryParseSecondsToNanoseconds`

### CODE REVIEW

File: deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java [hunk 1]
```diff
@@ -75,27 +84,77 @@ public void test_duration_to_header_value() {
...
+        "1.00000, 1000000000",
+        " 123.4567890246, 123456789024",
         "' 2. ', 2000000000",
         "3.0, 3000000000",
         "3.1, 3100000000",
         "1234567890.12345, 1234567890123450112",
         "1.523, 1523000000",
         "'   1.523  ', 1523000000",
+        "1234567890123467890123467890123467890123467890123467890123467890, 9223372036854775807",
+        "12345678901234678901234678901234678901234678901234678901234678901.123467890123467890, 9223372036854775807",
```
Comment (schlosna) [inline deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java:102]: we're implicitly capping max at `Long.MAX_VALUE`, may want to document edge cases

File: deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java [hunk 2]
```diff
@@ -75,27 +84,97 @@ public void test_duration_to_header_value() {
...
+            @ForAll @NumericChars @StringLength(min = 1, max = 100) String decimal) {
+        assertThat(Deadlines.tryParseSecondsToNanoseconds(integer))
+                .isNotNull()
+                .isGreaterThanOrEqualTo(0)
+                .isLessThanOrEqualTo(Long.MAX_VALUE);
+        assertThat(Deadlines.tryParseSecondsToNanoseconds(decimal))
+                .isNotNull()
+                .isGreaterThanOrEqualTo(0)
+                .isLessThanOrEqualTo(Long.MAX_VALUE);
+        ;
```
Comment (carterkozak) [inline deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java]:
```suggestion
```

File: deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java [hunk 3]
```diff
@@ -75,27 +84,97 @@ public void test_duration_to_header_value() {
...
+        assertThat(Deadlines.tryParseSecondsToNanoseconds(integer + '.' + decimal))
+                .isNotNull()
+                .isGreaterThanOrEqualTo(0)
+                .isLessThanOrEqualTo(Long.MAX_VALUE);
+        ;
```
Comment (carterkozak) [inline deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java]:
```suggestion
```

File: deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java (hunk 1 above)
Comment (bjlaub) [reply deadlines/src/test/java/com/palantir/deadlines/DeadlinesTest.java:102]: we could probably add something to the public javadoc on [Deadlines.parseFromRequest](https://github.com/palantir/deadlines-java/blob/develop/deadlines/src/main/java/com/palantir/deadlines/Deadlines.java#L120-L135) (that actually has no mention of what the type of the header value is intended to be, which is perhaps an oversight)

**Review by carterkozak — APPROVED** [review]
thanks!
